playbooks/ai-on-openlab/pre.yaml

```diff
@@ -0,0 +1,93 @@
+      set_fact:
+        user_file: "user_data_gpu"
+      when: is_gpu_enabled
+
+    - name: Shape login Demo User
+      shell:
+        cmd: |
+          set -x
+          set -e
+
+          useradd demo -p 'demo'
+          echo 'demo:demo' | chpasswd
+          usermod -s /bin/bash demo
+          mkdir /home/demo
+          chown demo:demo /home/demo
+          sed -i 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config
+          service sshd restart
+        executable: /bin/bash
+
+    - name: Upload Environment Info to Web
+      shell:
+        cmd: |
+          apt install -y python-pip
+          pip install pygithub
+
+          source_file=`ls "{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/" | grep moo-job`
+          job_name=$(echo $source_file | cut -d . -f1)
+          job_name=${job_name/_sleep/}
+          if {{ env_keeping | lower }};then
+            wget https://raw.githubusercontent.com/moo-ai/moo-ai.github.io/master/env_info/task_list_sleep.json -O env.json
+          else
+            wget https://raw.githubusercontent.com/moo-ai/moo-ai.github.io/master/env_info/task_list.json -O env.json
+          fi
+          cat > 'json_output.py' <<EOF
+          import json
+          exist_data = None
+          with open('env.json') as json_file:
+              exist_data = json.load(json_file)
+
+          ip_addr = "{{ nodepool.public_ipv4 }}"
+
+          exist_data["$job_name"] = {
+              "patch_set": "{{ zuul.patchset }}",
+          }
+          if '{{ env_keeping | lower }}' == 'true':
+              exist_data["$job_name"]["ip"] = ip_addr
+              exist_data["$job_name"]["credential"] = "demo/demo"
+          else:
+              result_url="http://logs.openlabtesting.org/logs/{{ zuul.change[-2:] }}/{{ zuul.change }}/{{ zuul.patchset }}/check/{{ zuul.job }}/{{ build_id.stdout }}/test_results/"
+              exist_data["$job_name"]["result_url"] = result_url
+
+          with open('env.json_tmp', 'w') as file:
+              json.dump(exist_data, file)
+          EOF
+
+          python json_output.py
+          cat env.json_tmp | python -m json.tool > env.json
+
+          cat > 'upload_env_info.py' <<EOF
+          import github
+          import sys
+
+          file_name = sys.argv[1]
+
+          with open(file_name) as f:
+              cont = f.read()
+
+          github_obj = github.Github('moo-ai', 'mooopenlab1')
+          repo = github_obj.get_user().get_repo('moo-ai.github.io')
+          env_file_name = 'env_info/task_list.json'
+          if '{{ env_keeping | lower }}' == 'true':
+              env_file_name = 'env_info/task_list_sleep.json'
+          try:
+              contents = repo.get_contents(env_file_name, ref="master")
+          except github.UnknownObjectException:
+              exit(1)
+
+          repo.update_file(contents.path, "upload $job_name ci job", cont, contents.sha, branch="master")
```

Comment:
how about just put these python scripts in separated files to make playbook clearer ?
my 2 cents.

Reply:
I'll move then into moo project later.